show-magnifier/src/main/java/com/esri/arcgismaps/sample/showmagnifier/components/ComposeMapView.kt

```diff
@@ -0,0 +1,92 @@
+/* Copyright 2023 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.showmagnifier.components
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.compose.ui.viewinterop.AndroidView
+import androidx.lifecycle.LifecycleOwner
+import com.arcgismaps.mapping.view.MapView
+import com.arcgismaps.mapping.view.SingleTapConfirmedEvent
+import kotlinx.coroutines.launch
+
+/**
+ * Wraps the MapView in a Composable function.
+ */
+@Composable
+fun ComposeMapView(
+    modifier: Modifier = Modifier,
+    mapViewModel: MapViewModel,
+    onSingleTap: (SingleTapConfirmedEvent) -> Unit = {}
+) {
+    // get an instance of the current lifecycle owner
+    val lifecycleOwner = LocalLifecycleOwner.current
+    // get an instance of the MapView state
+    val mapViewState = mapViewModel.mapViewState
+    // create and add MapView to the activity lifecycle
+    val mapView = createMapViewInstance(lifecycleOwner)
+
+    // wrap the MapView as an AndroidView
+    AndroidView(
+        modifier = modifier,
+        factory = { mapView },
+        // recomposes the MapView on changes in the MapViewState
+        update = { mapView ->
+            mapView.apply {
+                map = mapViewState.arcGISMap
+                setViewpoint(mapViewState.viewpoint)
+                // set the MapView's interaction options using the MapViewState
+                mapViewState.interactionOptions.apply {
+                    interactionOptions.isMagnifierEnabled = isMagnifierEnabled
+                    interactionOptions.allowMagnifierToPan = allowMagnifierToPan
+                }
```

Comment:
actually not a duplicate, as we are applying the mapViewState values to the mapView in update block.

its actually doing this 
```
            mapViewState.interactionOptions.apply {
                mapView.interactionOptions.isMagnifierEnabled = isMagnifierEnabled
                mapView.interactionOptions.allowMagnifierToPan = allowMagnifierToPan
            }
```

Reply:
removed `mapView.interactionOptions.allowMagnifierToPan = allowMagnifierToPan` as its not needed, as `allowMagnifierToPan ` property is always true

the magnifier and panning of the magnifier works with the changes

Reply:
I see, that makes sense, I misread this code. I actually think in this case it makes sense not to remove `allowMagnifierToPan`? Because perhaps the user would modify the main code and not modify this part.